Gems/EMotionFX/Code/EMotionFX/Source/MeshDeformerStack.cpp

```diff
@@ -110,10 +110,15 @@ namespace EMotionFX
         // if we have deformers in the stack
         const size_t numDeformers = m_deformers.size();
 
+        EMotionFX::SkinningInfoVertexAttributeLayer* skinLayer =
+            (SkinningInfoVertexAttributeLayer*)m_mesh->FindSharedVertexAttributeLayer(SkinningInfoVertexAttributeLayer::TYPE_ID);
+        const AZ::u32 numOrgVerts = skinLayer->GetNumAttributes();
+        const auto [numLocalJoints, highestJointId] = skinLayer->CalcLocalJointIndexCountAndHighestIndex(numOrgVerts);
+
         // iterate through the deformers and reinitialize them
         for (size_t i = 0; i < numDeformers; ++i)
         {
-            m_deformers[i]->Reinitialize(actor, node, lodLevel);
+            m_deformers[i]->Reinitialize(actor, node, lodLevel, highestJointId);
```

Comment:
What about just initializing the local joint array with the number of joints in the skeleton and calculate the numLocalJoints while filling it, resizing/shrinking it afterwards?

That way we don't have to iterate through the skin influences twice, inside CalcLocalJointIndexCountAndHighestIndex as well as Reinitialize()

Reply:
I've updated this to calculate it while building the mesh and just retrieve it using a Get function later.